lib/pure/memfiles.nim

```diff
@@ -245,6 +249,30 @@ proc open*(filename: string, mode: FileMode = fmRead,
       if close(result.handle) == 0:
         result.handle = -1
 
+proc flush*(f: var MemFile; attempts: int = 3) =
```

Comment:
Why would flushing require multiple attempts?

Reply:
Ahh, the hidden reviews contain the answers. The original code also contains this really useful comment which should be copied into our code IMO.